src/shapes.nim

```diff
@@ -0,0 +1,277 @@
+from std/fenv import epsilon
+from std/math import sgn, floor, sqrt, arccos, arctan2, PI
+import std/options
+
+import geometry, camera
+
+
+type
+    AABB* = tuple[min, max: Point3D]
+
+    ShapeKind* = enum
+        skAABox, skTriangle, skSphere, skPlane, skMesh
+
+    Shape* = object of RootObj
+        transf*: Transformation
+        aabb*: Option[AABB] = none(AABB)
+
+        case kind*: ShapeKind 
+        of skAABox: 
+            min, max: Point3D
+
+        of skTriangle: 
+            vertices*: tuple[A, B, C: Point3D]
+
+        of skMesh:
+            nodes*: seq[Point3D]
+            edges*: seq[tuple[left, right: int]]
```

Comment:
Here we have to change how we are working if we want to distinguish between triangular meshes and generic meshes: in a triangular mesh we don't need to give tuple in order to give the order in which vertices are connected, we just can use vertices indexes.